### PR TITLE
Reframe 4.x

### LINKED ISSUE #42: Allow apptainer as an alternative for singularity
We made the switch from singularity to apptainer. We still make the singularity command available, by symlinking to apptainer. However, `singularity --version` now gives `apptainer version 1.0.3-1`  which makes `VSCToolVersionTest_singularity` fail. Can we adapt the test so it also passes if `apptainer=1.0.3` is found?

### PR DESCRIPTION
**Pull request by Lewih**
fixes #42, #40.
fixes small bugs.
rename hortense in dodrio for consistency with its hostname 
Move to reframe 4.x

### CODE REVIEW

File: config_vsc.py [hunk 1]
```diff
@@ -1,29 +1,42 @@
 import grp
 import os
+import time
```
Comment (smoors) [inline config_vsc.py:3]: this import is not used?

File: config_vsc.py [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
-        'environ=%(check_environ)s',
-        'num_tasks=%(check_num_tasks)s',
-        'num_cpus_per_task=%(check_num_cpus_per_task)s',
-        'num_tasks_per_node=%(check_num_tasks_per_node)s',
-        'modules=%(check_modules)s',
-        'jobid=%(check_jobid)s',
-        'perf_var=%(check_perf_var)s',
-        'perf_value=%(check_perf_value)s',
-        'unit=%(check_perf_unit)s',
+standard_mode_options = [
```
Comment (smoors) [inline config_vsc.py:8]: this is specific for antwerpen, so can you rename it to e.g. `antwerpen_mode_options`?

File: config_vsc.py [hunk 3]
```diff
@@ -34,13 +47,13 @@
 for version in ['2018a', '2019b', '2021a']:
     genius_modulepath.append(f'/apps/leuven/skylake/{version}/modules/all')
 
-# Specify hortense access flag in order to run jobs
+# Specify dodrio access flag in order to run jobs
```
Comment (smoors) [inline config_vsc.py:50]: the official name is `hortense`, i think it's better to keep the official name?

File: config_vsc.py [hunk 4]
```diff
@@ -330,22 +343,12 @@
...
-                    'format': perf_logging_format,
-                    'append': True,
-                },
-            ],
         }
     ],
-}
+    'modes': [
+        {
+            'name': 'standard',
```
Comment (smoors) [inline config_vsc.py:350]: this mode is specific for antwerpen, so rename to e.g. `mode_antwerpen`?

File: run.sh [hunk 5]
```diff
@@ -1,11 +1,11 @@
-module load ReFrame/3.10.1
+module load ReFrame/4.1.0
```
Comment (smoors) [inline run.sh:1]: there is already a module for ReFrame 4.2.0, can we use that?

File: run.sh [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
-export RFM_OUTPUT_DIR=$HOME/reframe
-export RFM_PREFIX=$HOME/reframe
+export RFM_OUTPUT_DIR=$VSC_SCRATCH/reframe
+export RFM_PREFIX=$VSC_SCRATCH/reframe
 export RFM_CHECK_SEARCH_RECURSIVE=true
 export RFM_SAVE_LOG_FILES=true
 
-reframe --run  "$@"
-rm $(dirname $0)/reframe.out $(dirname $0)/reframe.log
+reframe -S strict_check=1 --run  "$@"
```
Comment (smoors) [inline run.sh:10]: `strict_check=true` is already the default, so can be removed?

File: tests/cue/env.py [hunk 7]
```diff
@@ -23,7 +23,8 @@ class VSCEnvTest(rfm.RunOnlyRegressionTest):
     def set_param(self):
         self.descr += self.envar
         exe = envars[self.envar]['exe']
-        self.executable = "python3 -c 'import os;{}'".format('\n'.join(exe))
+        # load Reframe to expose archspec in python path 
+        self.executable = "ml load ReFrame; python3 -c 'import os;{}'".format('\n'.join(exe))
```
Comment (smoors) [inline tests/cue/env.py:27]:
```suggestion
        self.executable = "ml ReFrame; python3 -c 'import os;{}'".format('\n'.join(exe))
```

File: config_vsc.py (hunk 1 above)
Comment (Lewih) [reply config_vsc.py:3]: ok

File: config_vsc.py (hunk 2 above)
Comment (Lewih) [reply config_vsc.py:8]: ok

File: config_vsc.py (hunk 3 above)
Comment (Lewih) [reply config_vsc.py:50]: ok

File: config_vsc.py (hunk 4 above)
Comment (Lewih) [reply config_vsc.py:350]: ok

File: run.sh (hunk 5 above)
Comment (Lewih) [reply run.sh:1]: Why not

**Comment by smoors** [comment]
in our config file, we should also change the name of the `'builtin'` environment, e.g. to `'default'`, to avoid this warning:
```
WARNING: redefinition of environment '*:builtin': already defined in '<builtin>'
```

**Comment by Lewih** [comment]
This fixes the warning.

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**